### DISCUSSION TITLE
Is it possible to say that there is a virus in this program?

### DISCUSSION THREAD

**Discussion by KetSpin** [winsiderss/systeminformer#1914] [Questions / Answers] [answered]
![image](https://github.com/winsiderss/systeminformer/assets/96338452/5261ee0d-274b-4e3f-abe7-290e0fd75e7f)
My antivirus when running this program referred to the virus: Trojan:Win32/Bearfoos.BE!ml
The test was done on the website: tria.ge

**Comment by KetSpin** [comment]
Please tell us what the data I provided in the photo means.

> ![образ](https://private-user-images.githubusercontent.com/96338452/286342825-5261ee0d-274b-4e3f-abe7-290e0fd75e7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExOTI1MDQsIm5iZiI6MTcwMTE5MjIwNCwicGF0aCI6Ii85NjMzODQ1Mi8yODYzNDI4MjUtNTI2MWVlMGQtMjc0Yi00ZTNmLWFiZTctMjkwZTBmZDc1ZTdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDE3MjMyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3OWIxNDc5ZDQ2MDRhMDllNGQ5YzI4ZGZmMDRjY2MzZDc0NjNiYmQzZGVjZjFlZjBkYjYwMzUzNTA2OGU2ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.a-3V_pneFgwr49r-XA9nyOI5S6wyOjA754CdWRerRKs) Мой антивирус при запуске этой программы ссылался на вирус: Trojan:Win32/Bearfoos.BE!ml Тест проводился на сайте: tria.ge

Please tell us what the data I provided in the photo means.

**Reply by jxy-s** [reply]
> Is it possible to say that there is a virus in this program?

No? That's up to you to decide. The entire project is open source. I would not classify it as malicious.

> My antivirus when running this program referred to the virus: Trojan:Win32/Bearfoos.BE!ml
> The test was done on the website: tria.ge

Seems like a false positive to me. The behaviors you listed I would not classify as malicious by themselves.

**Comment by Masamune3210** [comment]
Its not malicious, everything in that screenshot is things that are needed to do what SystemInformer supports doing. 
Here's a hint, if the virus definition has !ml in it, it means its a heuristic definition, aka machine learning.

**Comment by Masamune3210** [comment]
Btw, sites like that are wonderful for checking things out.....if you know what you are looking for. This is an example of a scary list of things a program did, looks malicious right?

It's a shell extension installer for a known clean shell extension. 
![image](https://github.com/winsiderss/systeminformer/assets/1053504/003850c9-f3f6-4e63-8d75-95a5847db325)

**Comment by dmex** [comment, marked as answer]
![image](https://github.com/winsiderss/systeminformer/assets/1306177/cd4ae980-243d-460f-8010-0d1ef867ed69)

The setup is an executable and writes an executable into  the ` \Program Files\SystemInformer ` directory. They could show the exact filename and directory where these files were "dropped" but they don't because these detections are broken and always incorrect.

![image](https://github.com/winsiderss/systeminformer/assets/1306177/0b90799e-1cf0-47be-9134-361d7cba7ed6)

This project is 15 years old. How does someone abuse Gitnub and Sourceforge for 15 years and nothing ever done about it?

![image](https://github.com/winsiderss/systeminformer/assets/1306177/aa37b6a9-e8d1-4b65-8f88-91f437f6db15)

We don't write anything into \System32?

![image](https://github.com/winsiderss/systeminformer/assets/1306177/4f1bce98-c947-4768-842c-052adff99bb9)

We don't write anything into \Windows?

![image](https://github.com/winsiderss/systeminformer/assets/1306177/7dcd8972-77fb-4ff0-b5dd-7962ad0bb56f)

We don't use ThreadHideFromDebugger? You can't hide from the debugger.

![image](https://github.com/winsiderss/systeminformer/assets/1306177/915614fe-a14b-47ce-a4fa-baeb6045b986)

We absolutely enumerate storage devices. Windows Task manager shows performance information for disk devices and we have the exact same feature. How are we expected to show disk statistics without first enumerating disk devices?

![image](https://github.com/winsiderss/systeminformer/assets/1306177/75e8834e-2ff2-469c-9113-b8edf5949f02)

![image](https://github.com/winsiderss/systeminformer/assets/1306177/6688cb6e-0805-4478-a850-47abd1ce2d7d)

I have no idea what this is. We don't modify the registry or query system information from the registry - All system information gets queried from [NtQuerySystemInformation](https://learn.microsoft.com/en-us/windows/win32/api/winternl/nf-winternl-ntquerysysteminformation) because that's where system information is located.

![image](https://github.com/winsiderss/systeminformer/assets/1306177/eff27b5a-4657-4308-b55e-ab4ecd8f5aa3)

A task manager querying processes is exactly what a task manager should be doing.

![image](https://github.com/winsiderss/systeminformer/assets/1306177/e4438a99-2d24-4900-9811-e919a46ea61e)

Since when did improving security become 'suspicious? 

a) A task manager should be reliable, and this feature is great for preventing legacy software from crashing the application. 

b) All downloads from the Windows Store are signed by Microsoft and every security product and kernel driver are also signed by Microsoft (including our binaries) - the policy does nothing to stop security products or third party software and is only effective with legacy software predating attestation signing. 

c) Anyone can enable/disable/override the policy using powershell: https://learn.microsoft.com/en-us/powershell/module/processmitigations/set-processmitigation?view=windowsserver2022-ps

![image](https://github.com/winsiderss/systeminformer/assets/1306177/d2993801-61a3-4303-8970-da3d844978c2)

Windows doesn't allow Task Managers to manage processes or query process information without various privileges - you've added privileges to your user account for programs to enable and they'll use them because they're enabled. 

> sites like that are wonderful for checking things out

These sites leave out critically relevant data and get abused to spread misleading information. We've had hundreds of threads and issues full of garbage produced by these websites over the last 3 years and all they've done is mislead people and cause problems.

**Comment by Masamune3210** [comment]
Like I said, they are good at what they do (not great, because they tend to misconstrue random actions as entirely different things), but without knowledge of how to read what it tells you, its literally just fear mongering r us
